server/models/Notification.js

```diff
@@ -40,6 +42,61 @@ export default function(Sequelize, DataTypes) {
     classMethods: {
       createMany: (notifications, defaultValues) => {
         return Promise.map(notifications, u => Notification.create(_.defaults({},u,defaultValues))).catch(console.error);
+      },
+
+      /**
+       * Get the list of subscribers to a mailing list (e.g. backers@:collectiveSlug.opencollective.com)
+       * For members, backers, and info: opt-in: we look for rows in the Notifications table
+       * For events, it's opt-out. We exclude people who have explicitly unsubscribed
+       */
+      getSubscribers: (collectiveSlug, mailinglist) => {
```

Comment:
That's exactly what I did with the sub functions `getSubscribersForEvent` and `getSubscribersForMailingList`.
It's just that I wanted to abstract that complexity from above. Because the only way to know that a given slug related to an event is to first look for an event for that slug.